assets/src/blocks/Media/MediaEditor.js

```diff
@@ -0,0 +1,150 @@
+          icon="format-image"
+          onSelect={ onSelectImage }
+          onError={ console.log }
+          accept="image/*"
+          allowedTypes={["image"]}
+        />
+      </PanelBody>
+    </InspectorControls>
+  ;
+}
+
+const renderView = (props, toAttribute) => {
+  const { attributes } = props;
+  const { video_title, description, embed_html, media_url, poster_url } = attributes;
+
+  return (
+    <Fragment>
+      <header>
+        <RichText
+          tagName="h2"
+          className="page-section-header"
+          placeholder={__('Enter title', 'planet4-blocks-backend')}
+          value={video_title}
+          onChange={toAttribute('video_title')}
+          keepPlaceholderOnFocus={true}
+          withoutInteractiveFormatting
+          characterLimit={40}
+          multiline="false"
+        />
+      </header>
+      <RichText
+        tagName="p"
+        className="page-section-description"
+        placeholder={__('Enter description', 'planet4-blocks-backend')}
+        value={description}
+        onChange={toAttribute('description')}
+        keepPlaceholderOnFocus={true}
+        withoutInteractiveFormatting
+        characterLimit={200}
+      />
+      {
+        (media_url && !media_url.endsWith('.mp4')) && !embed_html
+        ? <div className="block-edit-mode-warning components-notice is-error">
+            { __( 'The video URL could not be parsed.', 'planet4-blocks-backend' ) }
+          </div>
+        : <Fragment>
+            {
+              media_url && media_url.endsWith('.mp4')
+              ? <MediaElementVideo videoURL={ media_url } videoPoster={ poster_url } />
+              : <MediaEmbedPreview html={ embed_html || null } />
+            }
+          </Fragment>
+      }
+    </Fragment>
+  )
+}
+
+export const MediaEditor = (props) => {
+  const { attributes, setAttributes, isSelected } = props;
+  const { media_url, video_poster_img } = attributes;
+
+  // Assume that a non-URL is a YouTube video ID.
+  const parsedMediaURL = media_url && media_url.indexOf('/') === -1
+    ? `https://www.youtube.com/watch?v=${media_url}`
+    : media_url;
+
+  console.log('media URL', parsedMediaURL);
+
+  const toAttribute = attributeName => value => setAttributes({ [attributeName]: value });
+
+  if ( parsedMediaURL && parsedMediaURL.indexOf('youtube.com') > -1 ) {
+    parsedMediaURL.replace('youtube.com', 'youtube-nocookie.com');
+  }
+
+  const { embed_html, poster_url } = useSelect((select) => {
+    const embedPreview = select('core').getEmbedPreview( parsedMediaURL );
+    const media = select('core').getMedia(video_poster_img)
+    return {
+      embed_html: embedPreview ? embedPreview.html : null,
+      poster_url: media ? media.media_details.sizes.large.source_url : null
+    }
+  },
+  [ media_url, video_poster_img ]);
```

Comment:
Even though this `useSelect` is originally used to map something to props in the same way `withSelect` does, for  `useSelect` this is actually vestigial and not needed anymore.

It was previously mapped to props because the HOC paradigm works that way, but with hooks we actually don't need these intermediary props anymore since we're already inside of render. So we can avoid some duplication here:

```js
const [ embed_html, poster_url ] = useSelect(select => {
    const embedPreview = select('core').getEmbedPreview( parsedMediaURL );
    const media = select('core').getMedia(video_poster_img);

    return [
      embedPreview ? embedPreview.html : null,
      media ? media.media_details.sizes.large.source_url : null
    ]
  },
  [ media_url, video_poster_img ]
);

Reply:
Actually 2 separate `useSelect` calls is even more readable:
```js
  const embed_html = useSelect(select => {
    const embedPreview = select('core').getEmbedPreview(parsedMediaURL);

    return embedPreview ? embedPreview.html : null;
  }, [media_url]);

  const poster_url = useSelect(select => {
    const media = select('core').getMedia(video_poster_img);

    return media ? media.media_details.sizes.large.source_url : null;
  }, [video_poster_img]);